src/client/src/commons/form/profile/components/profileLayers/components/lithologicalDescriptionLayers/lithologicalDescriptionLayers.js

```diff
@@ -0,0 +1,447 @@
+import React, { useState, useEffect } from "react";
+import {
+  Box,
+  Button,
+  MenuItem,
+  FormControl,
+  Select,
+  InputLabel,
+  Input,
+  Stack,
+  TextField,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import ClearIcon from "@mui/icons-material/Clear";
+import ModeEditIcon from "@mui/icons-material/ModeEdit";
+import DeleteIcon from "@mui/icons-material/Delete";
+import AddCircleIcon from "@mui/icons-material/AddCircle";
+import WarningIcon from "@mui/icons-material/Warning";
+import {
+  updateLithologicalDescription,
+  useDomains,
+  deleteLithologicalDescription,
+} from "../../../../../../../api/fetchApiV2";
+import produce from "immer";
+import { useMutation, useQueryClient } from "react-query";
+import { withTranslation } from "react-i18next";
+
+const LithologicalDescriptionLayers = props => {
+  const {
+    isEditable,
+    lithologicalDescriptions,
+    setSelectedDescription,
+    selectedDescription,
+    layers,
+    addMutation,
+    selectedStratigraphyID,
+    t,
+  } = props;
+  const [fromDepth, setFromDepth] = useState(null);
+  const [toDepth, setToDepth] = useState(null);
+  const [description, setDescription] = useState(null);
+  const [qtDescriptionId, setQtDescriptionId] = useState(null);
+  const [displayDescriptions, setDisplayDescriptions] = useState(null);
+  const [showDeleteDialog, setShowDeleteDialog] = useState(0);
+
+  // React-query mutations and queries.
+  const queryClient = useQueryClient();
+  const domains = useDomains();
+
+  const deleteMutation = useMutation(
+    async id => {
+      const result = await deleteLithologicalDescription(id);
+      return result;
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries({ queryKey: ["lithoDesc"] });
+      },
+    },
+  );
+
+  const updateMutation = useMutation(
+    async params => {
+      const result = await updateLithologicalDescription(params);
+      return result;
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries({ queryKey: ["lithoDesc"] });
+      },
+    },
+  );
+
+  const selectableFromDepths = layers?.data?.map(l => l.depth_from);
+  const selectableToDepths = layers?.data?.map(l => l.depth_to);
+
+  useEffect(() => {
+    // Include empty items in description column to signal missing descriptions.
+    const tempDescriptions = [];
+    lithologicalDescriptions
+      .sort((a, b) => a.fromDepth - b.fromDepth)
+      .forEach((d, index) => {
+        if (
+          index !== 0 &&
+          d.fromDepth !== lithologicalDescriptions[index - 1].toDepth
+        ) {
+          tempDescriptions.push({
+            id: null,
+            fromDepth: lithologicalDescriptions[index - 1].toDepth,
```

Comment:
PP: `lithologicalDescriptions[index - 1].toDepth` in Variable mit aussagekräftigem Namen auslagern, das macht den Code lesbarer.